.editorconfig

```diff
@@ -0,0 +1,23 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_style = tab
+insert_final_newline = true
+
+[*.{cpp,hpp,c,h,mm}]
+trim_trailing_whitespace = true
```

Comment:
Out of curiosity, why isn't `trim_trailing_whitespace` applied to all file types?